docs/decisions/0005-migration-path-for-legacy-mechanisms.rst

```diff
@@ -0,0 +1,95 @@
+5. Migration Path for Legacy User Grouping Mechanisms
+#####################################################
+
+Status
+******
+
+**Draft** - 2025-06-03
+
+Context
+*******
+
+Open edX currently uses several user grouping mechanisms (Cohorts, Teams,
+Enrollment Track Groups), each with its own logic, storage, and integration
+points. This fragmentation results in:
+
+- Complicates maintenance and evolution.
+- Makes it difficult to implement new functionality.
+- Limits interoperability and extensibility.
+
+Two migration paths were evaluated to transition to a unified grouping system:
+
+- **Cross-System Synchronization**: creates an abstraction layer that
+  translates the new model to the legacy mechanisms.
+- **Behavior Replication**: builds a new and independent system that replicates
+  the observable behavior of the legacy mechanisms without integrating with
+  them.
+
+Decision
+********
+
+We select the behavior replication approach, eliminating direct dependencies on
+legacy systems. This choice enables a simpler, cleaner architecture with:
+
+- Full independence from legacy mechanisms from day one.
+- Elimination of complex synchronization or integration layers.
+- Reduced technical debt and maintenance costs during migration.
+
+Existing user-facing functionalities will be replicated in the new model, with
+migration executed in clear, isolated phases to minimize risk. Activation will
+be controlled via feature flags, configurable per course, organization, or
+platform.
+
+See `ADR 6 <docs/decisions/0006-replication-of-legacy-mechanisms-behavior.rst>`_
+for detailed rationale.
+
+Consequences
+************
+
+- The new system can evolve independently, allowing greater flexibility.
+- The responsibility for replicating legacy behavior lies entirely within the
+  new model, which must be thoroughly validated.
+- The transition can be carried out gradually, implementing one functionality
+  at a time, allowing individual behavior validation and more targeted testing.
+- Both new and legacy systems can coexist during rollout, avoiding user
+  disruption.
+- Legacy systems will be fully deprecated and removed post-transition,
+  improving maintainability and extensibility.
+
+Rejected Alternatives
+*********************
+
+Cross-System Synchronization
+============================
+
+This proposal involved creating a new unified model while maintaining indirect
```

Comment:
Should we specify that both approaches build on top of the new model but the difference is how they relate with legacy systems?